samples/OpenCVServer/OpenCVServer.java

```diff
@@ -0,0 +1,52 @@
+import java.io.*;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Vector;
+import java.net.*;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+
+public class OpenCVServer {
+    private ServerSocket serverSocket;
+    private Socket clientSocket;
+    private DataInputStream in;
+    private DataOutputStream out;
+
+    private BufferedImage currentImage;
+    private String fileName = "read.png";
+    private String resultFileName = "detected.png";
+ 
+    public void start(int port) {
+        try {
+            serverSocket = new ServerSocket(port);
+            clientSocket = serverSocket.accept();
+            out = new DataOutputStream(clientSocket.getOutputStream());
+            in = new DataInputStream(clientSocket.getInputStream());
+            currentImage = ImageIO.read(ImageIO.createImageInputStream(clientSocket.getInputStream()));
+            ImageIO.write(currentImage, "png", new File(fileName));
+            DetectMarker.detect(fileName);
+            currentImage = ImageIO.read(new File(resultFileName));
+            ImageIO.write(currentImage, "png", clientSocket.getOutputStream());
```

Comment:
There is no need to send back the whole image, you should only send detected markers locations (and sizes).